### PR TITLE
Add integration tests for new Imagetool command to create AuxImage

### PR DESCRIPTION
**Pull request by xiancao**
Add integration tests for new Imagetool command to create AuxImage

Jenkins result:
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/7439/
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/7440/
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/7456/
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/7457/
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/7458/

### CODE REVIEW

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/DomainUtils.java [hunk 1]
```diff
@@ -257,4 +265,78 @@ public static void deleteDomainResource(String domainNS, String domainUid) {
         domainUid,
         domainNS);
   }
+
+  /**
+   * Patch a domain with auxiliary image.
```
Comment (sankarpn) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/DomainUtils.java:270]:
```suggestion
   * Patch a domain with auxiliary image and verify pods are rolling restarted.
```

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/DomainUtils.java [hunk 2]
```diff
@@ -257,4 +265,78 @@ public static void deleteDomainResource(String domainNS, String domainUid) {
...
+    boolean aiPatched = assertDoesNotThrow(() ->
+            patchDomainCustomResource(domainUid, domainNamespace, patch, "application/json-patch+json"),
+        "patchDomainCustomResource(Auxiliary Image)  failed ");
+    assertTrue(aiPatched, "patchDomainCustomResource(auxiliary image) failed");
+
+    domain1 = assertDoesNotThrow(() -> getDomainCustomResource(domainUid, domainNamespace),
+        String.format("getDomainCustomResource failed with ApiException when tried to get domain %s in namespace %s",
+            domainUid, domainNamespace));
+    assertNotNull(domain1, "Got null domain resource after patching");
+    assertNotNull(domain1.getSpec(), domain1 + " /spec/serverPod is null");
```
Comment (sankarpn) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/DomainUtils.java:322]:
```suggestion
    assertNotNull(domain1.getSpec(), domain1 + " /spec is null");
```

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/DomainUtils.java [hunk 3]
```diff
@@ -257,4 +265,78 @@ public static void deleteDomainResource(String domainNS, String domainUid) {
...
+    assertNotNull(domain1.getSpec().getServerPod(), domain1 + " /spec/serverPod is null");
+    assertNotNull(domain1.getSpec().getServerPod().getAuxiliaryImages(),
+        domain1 + "/spec/serverPod/auxiliaryImages is null");
+
+    //verify the new auxiliary image in the new patched domain
+    auxiliaryImageList = domain1.getSpec().getServerPod().getAuxiliaryImages();
+
+    String auxiliaryImage = auxiliaryImageList.get(index).getImage();
+    getLogger().info("In the new patched domain imageValue is: {0}", auxiliaryImage);
```
Comment (sankarpn) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/DomainUtils.java:331]:
```suggestion
    getLogger().info("In the new patched domain, imageValue is: {0}", auxiliaryImage);
```

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiNewCreateAuxImage.java [hunk 4]
```diff
@@ -0,0 +1,617 @@
...
+          withStandardRetryPolicy,
+          createAuxiliaryImage(miiAuxiliaryImage1, modelList, archiveList),
+          logger,
+          "createAuxImage to be successful");
+    } else {
+      logger.info("!!!! auxiliary image {0}:{1} exists !!!!", miiAuxiliaryImage1, MII_BASIC_IMAGE_TAG);
+    }
+
+    // push auximage1 to repo for multi node cluster
+    if (!DOMAIN_IMAGES_REPO.isEmpty()) {
```
Comment (sankarpn) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiNewCreateAuxImage.java:173]: I believe dockerLoginAndPushImageToRegistry() itself does this check, please  verify.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiNewCreateAuxImage.java [hunk 5]
```diff
@@ -0,0 +1,617 @@
...
+    // create domain custom resource using auxiliary image
+    logger.info("Creating domain custom resource with domainUid {0} and auxiliary image {1}",
+        domain1Uid, miiAuxiliaryImage1);
+    Domain domainCR = createDomainResource(domain1Uid, domainNamespace,
+        WEBLOGIC_IMAGE_TO_USE_IN_SPEC, adminSecretName, OCIR_SECRET_NAME,
+        encryptionSecretName, replicaCount, "cluster-1", auxiliaryImagePath,
+        auxiliaryImageVolumeName, miiAuxiliaryImage1 + ":" + MII_BASIC_IMAGE_TAG);
+
+    // create domain and verify its running
+    logger.info("Creating domain {0} with auxiliary images {1} in namespace {2}",
```
Comment (sankarpn) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiNewCreateAuxImage.java:188]:
```suggestion
    logger.info("Creating domain {0} with auxiliary image {1} in namespace {2}",
```

**Review by sankarpn — CHANGES_REQUESTED** [review]
minor corrections

**Review by maggiehe00 — COMMENTED** [review]
It looks the new test class is ready to be run on OKD. If you will be on vacation next Monday today you probably do not have enough time to run complete OKD test suite on Jenkins (some other test is running) and add the new test class into wko-okd-wls-cert profile. 
It will be good if you can open a track JIRA to confirm the above.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/DomainUtils.java (hunk 1 above)
Comment (xiancao) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/DomainUtils.java]: fixed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/DomainUtils.java (hunk 2 above)
Comment (xiancao) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/DomainUtils.java]: fixed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/DomainUtils.java (hunk 3 above)
Comment (xiancao) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/DomainUtils.java]: fixed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiNewCreateAuxImage.java (hunk 4 above)
Comment (xiancao) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiNewCreateAuxImage.java]: fixed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiNewCreateAuxImage.java (hunk 5 above)
Comment (xiancao) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiNewCreateAuxImage.java]: fixed

**Comment by xiancao** [comment]
I will create a new jira task to add this test to wko-okd-wls-cert.

**Review by maggiehe00 — COMMENTED** [review]
Please add new JIRA number here as reference.

**Review by sankarpn — APPROVED** [review]
LGTM

**Review by maggiehe00 — APPROVED** [review]
LGTM

**Comment by xiancao** [comment]
created a jira (#94894) to add the test to wko-okd-wls-cert profile.

**Pull request merged**